back_end/src/apis/major.py

```diff
@@ -31,7 +31,10 @@ def create_major():
 @major_api_bp.route('/get_majors', methods=['GET'])
 def get_majors():
     majors = Major.get_majors()
-    majors = list(map(lambda x: x.to_json(), majors))
+    if majors:
+        majors = list(map(lambda x: x.to_json(), majors))
+    else:
```

Comment:
Don't think this else is necessary since if the majors at line 33 is empty then at line 34 the list() will just give an empty list.
Did you encounter a bug without this else? Since my code did not have this else so if you do have a bug without it then pls tell me so that I can change my code as well.

Reply:
I will leave Benson to answer this question, but I will merge his current PR so that David can test out the foreign key for requirements